### PR TITLE
char MsgText[<length>] should not be globally defined from header file

### LINKED ISSUE #207: char MsgText[<length>] should not be globally defined from header file
The following char array should not be instantiated from a header file, it should be in the ``main()`` function as needed to be correctly scoped.

```c
/* Standard string buffer for use in messages */
char MsgText[SZ_LINE+1];
```

This is true for both [calacs](https://github.com/spacetelescope/hstcal/blob/master/pkg/acs/calacs/include/acs.h#L39-L40) and [calwf3](https://github.com/spacetelescope/hstcal/blob/master/pkg/wfc3/calwf3/include/msg.h#L10-L11).

### Update:
I commend this up into ``/include/hstcal.h`` as part of #191 since it is used in ``trlbuf.c``.  This still needs to be moved out of the header file before the PR for #191 is merged.  __Update:__ Done.

### PR DESCRIPTION
**Pull request by jamienoss**
resolves #207

### CODE REVIEW

**Comment by jamienoss** [comment]
At present, this PR does not expand the inclusions of hstcal.h to all necessary individual files. Instead, it is 'broadcast' via the include of acs.h & wf3.h. Both of these files need to include hstcal.h anyhow, for the inclusion of CHAR_FNAME_LENGTH, so... This is not a solid argument, I'm looking into a refactoring script to automate this change. Please review as is for now.

**Comment by jamienoss** [comment]
@sosey @pllim I am not sure if you have time to review this in full but wanted it on your radar in case you could think of any thing that I may have missed.

**Comment by jamienoss** [comment]
I have not squashed the commits to significantly aid in this review. Please review each commit by itself and in turn.

File: pkg/acs/calacs/include/acs.h [hunk 1]
```diff
@@ -94,7 +90,7 @@ void errchk ();                 /* HSTIO error check */
 
 /* A reference image. */
 typedef struct {
-    char name[ACS_LINE];            /* name of image */
+    char name[CHAR_BUFFER_LENGTH];            /* name of image */
```
Comment (jamienoss) [inline pkg/acs/calacs/include/acs.h]: For this commit, this should still be ``ACS_LINE``.

File: pkg/wfc3/calwf3/include/wf3.h [hunk 2]
```diff
@@ -85,7 +86,7 @@ typedef enum SwitchVals_ SwitchVals;
 
 /* A reference image. */
 typedef struct {
-    char name[SZ_LINE+1];            /* name of image */
+    char name[CHAR_BUFFER_LENGTH+1];            /* name of image */
```
Comment (jamienoss) [inline pkg/wfc3/calwf3/include/wf3.h]: For this commit, this should still be ``ACS_LINE``.

File: pkg/wfc3/calwf3/include/wf3.h [hunk 3]
```diff
@@ -96,7 +97,7 @@ typedef struct {
 
 /* A reference table. */
 typedef struct {
-    char name[SZ_LINE+1];            /* name of table */
+    char name[CHAR_BUFFER_LENGTH+1];            /* name of table */
```
Comment (jamienoss) [inline pkg/wfc3/calwf3/include/wf3.h]: For this commit, this should still be ``SZ_LINE``.

File: pkg/acs/calacs/include/acs.h [hunk 4]
```diff
@@ -103,7 +99,7 @@ typedef struct {
 
 /* A reference table. */
 typedef struct {
-    char name[ACS_LINE];            /* name of table */
+    char name[CHAR_BUFFER_LENGTH];            /* name of table */
```
Comment (jamienoss) [inline pkg/acs/calacs/include/acs.h]: For this commit, this should still be ``ACS_LINE``.

File: include/hstcal.h [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+#define HSTCAL_INCL
+
+// Note: Use of CHAR_FNAME_LENGTH & CHAR_LINE_LENGTH are interchanged throughout and should therefore be identical (#164).
+// However, for now, keep them separated such that they can be altered back to their original IRAF equivalents set in include/c_iraf.h:
+// # define IRAF_SZ_LINE    1023
+// # define IRAF_SZ_FNAME    255
+#define CHAR_FNAME_LENGTH 255
+#define CHAR_LINE_LENGTH 255
+
+#define MSG_BUFF_LENGTH CHAR_BUFFER_LENGTH + 1
```
Comment (jamienoss) [inline include/hstcal.h]: For this commit, this should be ``CHAR_LINE_LENGTH``.

**Comment by jamienoss** [comment]
The ultimate test is to checkout the branch and make sure that it builds, compiles and links.

**Comment by jamienoss** [comment]
@mdlpstsci thanks.

**Comment by jamienoss** [comment]
@sosey & @pllim are either of you able to look at this or should I just merge?

**Comment by pllim** [comment]
No, I don't have anything to say about this. I trust Michele's judgement.

**Comment by jamienoss** [comment]
@sosey thanks, will do.

**Comment by jamienoss** [comment]
@mdlpstsci please review the last commit added.

**Comment by jamienoss** [comment]
@mdlpstsci -  commits squashed, one final review approval needed, thanks.

**Comment by jamienoss** [comment]
@mdlpstsci I got my include script working so I fixed the indirect includes for hstcal.h.

**Comment by jamienoss** [comment]
@mdlpstsci last review, I promise. Commits squashed.

**Pull request merged**